docs/source/examples/zne_logical_rb_cirq_stim.md

```diff
@@ -0,0 +1,307 @@
+        if instruction.name != "TICK":
+            noisy.append(
+                "X_ERROR",
+                instruction.targets_copy(),
+                l_err,
+            )
+            noisy.append(
+                "Z_ERROR",
+                instruction.targets_copy(),
+                l_err,
+            )
+    return noisy
+```
+
+## Execute circuits
+
++++
+
+The Stim executor defined below accepts a Stim circuit, to which it applies the noise model, and then compiles and executes.
+For more information on Mitiq-compatible executors, see the [Executors](../guide/executors.md) section of the Mitiq user guide.
+
+```{code-cell} ipython3
+def stim_executor(circuit, p_err, p_th, distance, shots):
+    noisy_stim_circ = add_noise_to_stim_circuit(circuit, p_err, p_th, distance)
+    sampler = noisy_stim_circ.compile_sampler()
+    result = sampler.sample(shots=shots)
+    counts = np.count_nonzero(
+        [r[0].astype(int) + r[1].astype(int) for r in result]
+    )
+    return 1 - counts / shots
+```
+
+We will simulate the RB circuits at six different code distances and corresponding logical error rates.
+The distance of the error correcting code parameterizes the size of the physical data block per logical qubit and is directly related to the number of errors the code can correct.
+
+```{code-cell} ipython3
+code_distances = range(21, 11, -2)
+```
+
+Given a fixed (serial) sampling budget, here `base_shots = 10,000`, at code distances below the maximum available on the device, we can parallelize executions of each logical RB circuit and thereby take additional samples simultaneously.
+Note that we are not cutting the RB circuits but instead virtually splitting up the device to run the same logical RB circuit instance multiple times in parallel, with fewer qubits used for correction.
+The function `scale_shots` calculates how many total samples we can take, assuming lower code distance executions can be parallelized.
+Fixing a sampling budget across unmitigated and mitigated workflows provides a fair comparison of the results in terms of the resources required.
+
+```{code-cell} ipython3
+def scale_shots(
+    num_device_qubits, scaled_distance, base_shots, n_qubits_circuit
+):
+    used_qubits = n_qubits_circuit * scaled_distance**2
+    return base_shots * int(num_device_qubits / used_qubits)
+```
+
+To minimize execution time, we convert each trial circuit only once, outside of the execution loop.
+
+```{code-cell} ipython3
+noisy_results = np.zeros((trials, len(code_distances)))
+stim_circuits = [
+    stimcirq.cirq_circuit_to_stim_circuit(c) for c in filled_circuits
+]
+
+for i, code_distance in enumerate(code_distances):
+    for t in range(trials):
+        noisy_results[t, i] = stim_executor(
+            stim_circuits[t],
+            p_err,
+            p_th,
+            code_distances[i],
+            scale_shots(device_size, code_distances[i], 4 * base_shots, n_qubits),
+        )
```

Comment:
You've already got the `code_distance` from the for loop. No need to fetch it again! You might be able to do something similar with the inner loop going over `stim_circuits` as opposed to `range(trials)`, but I'm not confident `len(stim_circuits) = trials`.
```suggestion
for i, code_distance in enumerate(code_distances):
    for t in range(trials):
        noisy_results[t, i] = stim_executor(
            stim_circuits[t],
            p_err,
            p_th,
            code_distance,
            scale_shots(device_size, code_distance, 4 * base_shots, n_qubits),
        )
```

Reply:
I think this comment applies where you converted another for loop to use `enumerate` as well.

Reply:
Good catch, and indeed `trials` is the same as `len(stim_circuits)`, so I simplified that as well.